src/include/sof/schedule/ll_schedule.h

```diff
@@ -29,9 +29,9 @@
 	tracev_event(TRACE_CLASS_SCHEDULE_LL, format, ##__VA_ARGS__)
 
 #define ll_sch_set_pdata(task, data) \
-	(task->private = data)
+	do { (task)->private = data; } while (0)
```

Comment:
Do we really need do .. while for single line statements?

Reply:
@tlauda yes. This is a function-like macro, enclosing it in parentheses as above makes no sense. You never do
```
(x = 0);
```
somewhere in the middle of your code. Then also `task` had to be enclosed in parentheses, data too. So, you could write
```
#define ll_sch_set_pdata(task, data) (task)->private = (data)
```
but that wouldn't be very safe either, because then you could write something like `pointer->ll_sch_set_pdata(task, data)` which accidentally might then compile, but that's not what is wanted. You really want function-like semantics here.